src/sagemaker/jumpstart/types.py

```diff
@@ -311,6 +312,73 @@ def to_json(self) -> Dict[str, Any]:
         return json_obj
 
 
+class JumpStartInstanceTypeVariants(JumpStartDataHolderType):
+    """Data class for JumpStart instance type variants."""
+
+    __slots__ = [
+        "regional_aliases",
+        "variants",
+    ]
+
+    def __init__(self, spec: Optional[Dict[str, Any]]):
+        """Initializes a JumpStartInstanceTypeVariants object from its json representation.
+
+        Args:
+            spec (Dict[str, Any]): Dictionary representation of instance type variants.
+        """
+        self.from_json(spec)
+
+    def from_json(self, json_obj: Optional[Dict[str, Any]]) -> None:
+        """Sets fields in object based on json.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of instance type variants.
+        """
+
+        if json_obj is None:
+            return
+
+        self.regional_aliases: dict = json_obj["regional_aliases"]
+        self.variants: dict = json_obj["variants"]
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartInstanceTypeVariants object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+    def get_image_uri(self, instance_type: str, region: str) -> Optional[str]:
+        """Returns image uri from instance type and region.
+
+        Returns None if no instance type is available or found.
```

Comment:
add a Raises section please

Reply:
update return type to `Optional[Dict[str, Any]]` please

Reply:
It does not return a dictionary, so keeping as is.